### ISSUE TITLE
[OOAnalyzer] Lots of bad operand size found

### ISSUE THREAD

**Issue by ldesgoui** [cmu-sei/pharos#71]
Hello,

I'm currently running OOAnalyzer on a x86 32 bit Linux shared library (~30 megabytes) and after a slow but error-free partition phase, the analysis phase is returning a lot of these errors:

```
OOAN[ERROR]: bad operand size found @ 0x00732717 : 0
OOAN[ERROR]: bad operand size found @ 0x0073271D : 0
OOAN[ERROR]: bad operand size found @ 0x00732745 : 0
OOAN[ERROR]: bad operand size found @ 0x0073276A : 0
OOAN[ERROR]: bad operand size found @ 0x00732770 : 0
OOAN[ERROR]: bad operand size found @ 0x007327A6 : 0
```

According to [this](https://github.com/cmu-sei/pharos/blob/9608374b2f02b9f6c57bd24a823c7ef7012ed2a9/libpharos/funcs.cpp#L1265-L1273), it appears to be unhandled right now, would that mean that my analysis is for naught? (If you have the chance to reply soon, I would be grateful, I'm currently renting server space for this, if not, that's completely okay) Would you be interested in analyzing the library yourselves? (Although I don't believe it's legal for me to redistribute it, it is free to acquire)

Thank you.

**Comment by sei-eschwartz** [comment]
I'm not sure what's going on with that particular error.  @sei-ccohen any ideas?  But it would be helpful to know what library it is in.

But the more pressing issue is that OOAnalyzer really does not support Linux binaries.  It's hard to say what kind of results you will get, but probably not very good.  If it was me, I wouldn't bother renting server space for this.

We've had very good luck recently on large Windows executables.  So if you can happen to find a Windows version you'd be in much better shape.

**Comment by cfcohen** [comment]
This "error" (really more of a warning) is in the function hashing code, and shouldn't affect your OOAnalyzer results at all.  In general, OOAnalyzer will attempt to cope with whatever problems it encounters, and still try to produce a result.  So that's good news, but as sei-eschwartz just pointed out, OOAnalyzer doesn't really support Linux executables.   I also would recommend cancelling your job and not renting space for this analysis.

Also, when running (windows) executables of this size, you'll probably also want to break things into several steps.   One trick is to specify --serialize so that the partioning step can be be restarted more quickly, but that requires a fair bit of disk space.  Another is to specify --prolog-facts to just generate the initial facts, and then run the Prolog step using the recently committed SWI Prolog scripts.  This approach doesn't have a very user-friendly interface just yet -- for example, there's currently no way to convert the Prolog results back into JSON for import into Ghidra or IDA Pro.  :-(  We're working on that.  But for very large programs, it's worth figuring out how to run OOAanlzyer this way, since the job might run for a couple dozen hours or more.

In general, it usually work gaining some experience on an executable that's <1MB in size before trying a file this big, just so you know what the entire process is, but (again for Windows executables), we've recently been succeeding on some files of this size.

**Comment by ldesgoui** [comment]
Thank you for the very quick responses, I appreciate it very much.

It's the main binary component of a game server that I'm analysing for some behaviors (the file in question is `Steam/steamapps/common/Team Fortress 2 Dedicated Server/tf/bin/server_srv.so`, which would unfortunately require you to use steamcmd to download).
I did not realize OOAnalyzer support those, everything seemed to go fine actually!
I can thankfully switch to the Windows version of this binary, which I may try tomorrow.
I've only had to rent a VM because the partition would grow past my 16gb available, but the pricing is per-minute and considerably small so no big losses there.

Thank you for all the explanations, have a nice week.
